### ISSUE TITLE
MDV not properly updated when filesystem is auto-extended

### ISSUE THREAD

**Issue by ingothierack** [stratis-storage/stratisd#1474]
Strange behavior now here.

First Problem with autoextending the FS from 1TB to 2 TB see #1456.
Restart of stratisd daemon after fs was full extend it to 2 TB(no autoextending).
Do fill up the fs to 1.3 TB, works till now.

Reboot the server, it fails now on mount this filesystem with invalid superblock. 

If it check the fs, it is set to only 1 TB size

from lsblk
stratis-1-210eb8dac6cb440993454dac6a885dae-thin-fs-f4a9283f89c3457bb5980d94eb82cc34 253:37   0     1T  0 stratis 

output stratis fs list

stratis_pool01  fs_share01       1.22 TiB     Mar 15 2019 14:41  /stratis/stratis_pool01/fs_share01       f4a9283f89c3457bb5980d94eb82cc34  

from logfile stratisd start output
```
Mar 27 21:42:03 nas001 stratisd[11431]:                        "fs_share01",
Mar 27 21:42:03 nas001 stratisd[11431]:                        SimpleRef(
Mar 27 21:42:03 nas001 stratisd[11431]:                            f4a9283f-89c3-457b-b598-0d94eb82cc34
Mar 27 21:42:03 nas001 stratisd[11431]:                        )
Mar 27 21:42:03 nas001 stratisd[11431]:                    ): StratFilesystem {
Mar 27 21:42:03 nas001 stratisd[11431]:                        thin_dev: ThinDev {
Mar 27 21:42:03 nas001 stratisd[11431]:                            dev_info: DeviceInfo {
Mar 27 21:42:03 nas001 stratisd[11431]:                                hdr: Struct_dm_ioctl {
Mar 27 21:42:03 nas001 stratisd[11431]:                                    version: [
Mar 27 21:42:03 nas001 stratisd[11431]:                                        4,
Mar 27 21:42:03 nas001 stratisd[11431]:                                        39,
Mar 27 21:42:03 nas001 stratisd[11431]:                                        0
Mar 27 21:42:03 nas001 stratisd[11431]:                                    ],
Mar 27 21:42:03 nas001 stratisd[11431]:                                    data_size: 305,
Mar 27 21:42:03 nas001 stratisd[11431]:                                    data_start: 312,
Mar 27 21:42:03 nas001 stratisd[11431]:                                    target_count: 1,
Mar 27 21:42:03 nas001 stratisd[11431]:                                    open_count: 0,
Mar 27 21:42:03 nas001 stratisd[11431]:                                    flags: 32,
Mar 27 21:42:03 nas001 stratisd[11431]:                                    event_nr: 0,
Mar 27 21:42:03 nas001 stratisd[11431]:                                    dev: Device {
Mar 27 21:42:03 nas001 stratisd[11431]:                                        major: 253,
Mar 27 21:42:03 nas001 stratisd[11431]:                                        minor: 37
Mar 27 21:42:03 nas001 stratisd[11431]:                                    },
Mar 27 21:42:03 nas001 stratisd[11431]:                                    name: "stratis-1-210eb8dac6cb440993454dac6a885dae-thin-fs-f4a9283f89c3457bb5980d94eb82cc34",
Mar 27 21:42:03 nas001 stratisd[11431]:                                    uuid: "stratis-1-210eb8dac6cb440993454dac6a885dae-thin-fs-f4a9283f89c3457bb5980d94eb82cc34"
Mar 27 21:42:03 nas001 stratisd[11431]:                                }
Mar 27 21:42:03 nas001 stratisd[11431]:                            },
Mar 27 21:42:03 nas001 stratisd[11431]:                            table: ThinDevTargetTable {
Mar 27 21:42:03 nas001 stratisd[11431]:                                table: TargetLine {
Mar 27 21:42:03 nas001 stratisd[11431]:                                    start: Sectors(0),
Mar 27 21:42:03 nas001 stratisd[11431]:                                    length: Sectors(2147483648),
Mar 27 21:42:03 nas001 stratisd[11431]:                                    params: ThinTargetParams {
Mar 27 21:42:03 nas001 stratisd[11431]:                                        pool: Device {
Mar 27 21:42:03 nas001 stratisd[11431]:                                            major: 253,
Mar 27 21:42:03 nas001 stratisd[11431]:                                            minor: 20
Mar 27 21:42:03 nas001 stratisd[11431]:                                        },
Mar 27 21:42:03 nas001 stratisd[11431]:                                        thin_id: ThinDevId {
Mar 27 21:42:03 nas001 stratisd[11431]:                                            value: 16
Mar 27 21:42:03 nas001 stratisd[11431]:                                        },
Mar 27 21:42:03 nas001 stratisd[11431]:                                        external_origin_dev: None
Mar 27 21:42:03 nas001 stratisd[11431]:                                    }
Mar 27 21:42:03 nas001 stratisd[11431]:                                }
Mar 27 21:42:03 nas001 stratisd[11431]:                            }
Mar 27 21:42:03 nas001 stratisd[11431]:                        },
```
[Wed Mar 27 21:12:52 2019] attempt to access beyond end of device
[Wed Mar 27 21:12:52 2019] dm-37: rw=4096, want=4294967296, limit=2147483648

Looks to me, it can't access to the end of fs to get the superblock due to limit to 1 TB.

How to repair this???????

I'm on stratis 1.0.3, Server is Fedora 29
Linux nas001 5.0.3-200.fc29.x86_64 #1 SMP Tue Mar 19 15:07:58 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

**Comment by mulkieran** [comment]
bz: https://bugzilla.redhat.com/show_bug.cgi?id=1696876.

**Comment by mulkieran** [comment]
Closed by: #1480.
